indico/queries/forms.py

```diff
@@ -0,0 +1,60 @@
+from typing import List
+from pathlib import Path
+
+from indico.client.request import RequestChain, GraphQLRequest
+from indico.queries.storage import UploadBatched, UploadDocument
+from indico.queries.jobs import Job
+
+class _FormPreprocessing(GraphQLRequest):
+
+    query = """
+        mutation($files: [FileInput]) {
+            activeFormFields(
+                files: $files
+            ) {
+                jobIds
+            }
+        }
+    """
+
+    def __init__(self, files):
+        super().__init__(
+            query=self.query, variables={"files": files}
+        )
+
+    def process_response(self, response):
+        jobs = super().process_response(response)["activeFormFields"]["jobIds"]
+        if jobs:
+            return [Job(id=j) for j in jobs]
+        else:
+            return []
+
+
+# TODO: move into indico-client
+class FormPreprocessing(RequestChain):
+    """
+    Attempt to auto-detect form fields and labels
+
+    Args:
+        files (str): list of filepaths to upload
+
+    Returns:
+        suggested_fields: list of dictionarys (1 per PDF) containing structured field data 
+    """
+
+    def __init__(
```

Comment:
Do we want to add a wait kwargs here like the other jobs?

Reply:
Seems warranted.

Reply:
Ok on second glance it seems like none of the other queries that return jobs do this because JobStatus is the query with a wait kwarg.

Reply:
If I added a wait kwarg would I still return jobs (just jobs that we already know are done / we have blocked on)?  Or have this be polymorphic and return job results when wait=True?

Reply:
OH right yeah right... hum I wrote it that way I should know ;p